libraries/chain/controller.cpp

```diff
@@ -4312,8 +4312,17 @@ struct controller_impl {
          if (qc) {
             verify_qc_future.get();
          }
-         integrate_received_qc_to_block(bsp); // Save the received QC as soon as possible, no matter whether the block itself is valid or not
-         consider_voting(bsp, use_thread_pool_t::no);
+         if (async_voting == async_t::yes) {
+            boost::asio::post(thread_pool.get_executor(), [this, bsp=bsp]() {
+               try {
+                  integrate_received_qc_to_block(bsp); // Save the received QC as soon as possible, no matter whether the block itself is valid or not
+                  consider_voting(bsp, use_thread_pool_t::no);
```

Comment:
Why `no` here in `use_thread_pool_t::no`?

Reply:
Because we are already posting into the `thread_pool`, no reason to post it again into the same thread_pool`.